nordicsemi/dfu/dfu_transport_ble.py

```diff
@@ -231,7 +231,9 @@ def close(self):
             raise IllegalStateException('DFU Adapter is already closed')
         super(DfuTransportBle, self).close()
         self.dfu_adapter.close()
-        self.dfu_adapter = None
+        if self.dfu_adapter is not None:
```

Comment:
Yes, you're right. There is a mistake here. I will try to investigate for more details